### PR TITLE
Docs of new JS SDK features

### PR DESCRIPTION
**Pull request by adams85**
### Description

Docs of new JS SDK features

### Requirement checklist

- [x] I have validated my changes on a test/local environment.
- [ ] I have checked the SNYK/Dependabot reports and applied the suggested changes.
- [ ] (Optional) I have updated outdated packages.

### CODE REVIEW

File: website/docs/sdk-reference/js.md [hunk 1]
```diff
@@ -378,89 +468,101 @@ ConfigCat - INFO - Evaluate 'isPOCFeatureEnabled'
...
-Config's data stored in a cache, it is efficiency increases in retrieval of data and performance of the client. If you would like to use your cache solution (for example your system uses external or distributed cache) you can implement those function and set to `cache` parameters in the setting.
+Config data is stored in a cache for reducing network traffic and for improving performance of the client. If you would like to use your own cache solution (for example when your system uses external or distributed cache) you can implement the [`ICache` interface](https://github.com/configcat/common-js/blob/master/src/Cache.ts) and set the `cache` parameter in the options.
 
 ```js
-function myCustomCache() {}
+class MyCustomCache {
+  set(key, config) {
+    // `key` [string] - a unique cache key
+    // `config` [object (ProjectConfig)] - configcat's cache config item
+    // insert here your cache write logic
```
Comment (sigewuzhere) [inline website/docs/sdk-reference/js.md:533]:
```suggestion
    // insert your cache write logic here
```

File: website/docs/sdk-reference/js.md [hunk 2]
```diff
@@ -378,89 +468,101 @@ ConfigCat - INFO - Evaluate 'isPOCFeatureEnabled'
...
+  }
+  get(key) {
+    // `key` [string] - a unique cache key
+    // insert here your cache read logic
```
Comment (sigewuzhere) [inline website/docs/sdk-reference/js.md:537]:
```suggestion
    // insert your cache read logic here
```

File: website/docs/sdk-reference/js.md [hunk 3]
```diff
@@ -378,89 +468,101 @@ ConfigCat - INFO - Evaluate 'isPOCFeatureEnabled'
...
 
 ## Using custom cache implementation
 
-Config's data stored in a cache, it is efficiency increases in retrieval of data and performance of the client. If you would like to use your cache solution (for example your system uses external or distributed cache) you can implement those function and set to `cache` parameters in the setting.
+Config data is stored in a cache for reducing network traffic and for improving performance of the client. If you would like to use your own cache solution (for example when your system uses external or distributed cache) you can implement the [`ICache` interface](https://github.com/configcat/common-js/blob/master/src/Cache.ts) and set the `cache` parameter in the options.
 
 ```js
-function myCustomCache() {}
+class MyCustomCache {
+  set(key, config) {
```
Comment (sigewuzhere) [inline website/docs/sdk-reference/js.md:530]: This 'config' might be a chance for a misunderstanding. As I'm not sure if this is some kind of configuration of the cache or is this the config that's being cached. I suggest using config only when trying to reference the config.json.

**Review by sigewuzhere — CHANGES_REQUESTED** [review]

File: website/docs/sdk-reference/js.md (hunk 3 above)
Comment (adams85) [reply website/docs/sdk-reference/js.md:530]: There is a comment a few lines below which describes the meaning of this parameter, so I wouldn't change this unless you insist.

**Review by laliconfigcat — DISMISSED** [review]

**Review by laliconfigcat — APPROVED** [review]

**Pull request merged**